File: Logic_HandsOn/C++/Problems on bit,File Handing,Pattern etc/program171.cpp
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a Java program which accept string from user and count number of capital characters.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

#include<iostream>
#include<stdio.h>

using namespace std;

class StringInput
{
	public:

		char Arr[30];

		void Accept()
		{
			cout<<"Enter the String: ";
			scanf("%[^'\n']s",Arr);
		}

};

class StringDemo:public StringInput
{
	public:

		int CountCapital()
		{
			char *str = Arr;
			int iCnt=0;

			while(*str!='\0')
			{
				if(*str>='A' && *str<='Z')
				{
					iCnt++;
				}
				str++;
			}

			return iCnt;
		}
};

int main()
{
	
	int iRet = 0;

	StringDemo *sObj = new StringDemo();
	sObj->Accept();	
	iRet = sObj->CountCapital();

	cout<<"Output: "<<iRet<<"\n";
}

/*
OUTPUT:

Enter the String: Vishal PAtil
Output: 3

*/

File: Logic_HandsOn/Java/Problems on bit,file handling,String etc/Program171.java
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
//
//	Program:
//		Write a Java program which accept string from user and count number of capital characters.
//
////////////////////////////////////////////////////////////////////////////////////////////////////////////////////

import java.util.Scanner;

class StringDemo
{
	public int CountCapital(String str)
	{
		int iCnt=0,iCount=0;
		while(iCnt!=str.length())
		{
			if(str.charAt(iCnt)>='A' && str.charAt(iCnt)<='Z')
			{
				iCount++;
			}
			iCnt++;
		}
		
		return iCount;
	}
}


class Program
{
	public static void main(String arg[])
	{
		int iRet = 0;

		Scanner sObj = new Scanner(System.in);
		System.out.println("Input: ");
		String str = sObj.nextLine();
	
		StringDemo sobj = new StringDemo();
		iRet = sobj.CountCapital(str);
	
		System.out.println(iRet+" ");
	}
}

/*
OUTPUT:

Input:
Vishal PAtil
3

*/
